lib/ansible/modules/cloud/amazon/aws_az_facts.py

```diff
@@ -0,0 +1,132 @@
+
+DOCUMENTATION = '''
+module: aws_az_facts
+short_description: Gather facts about availability zones in AWS.
+description:
+    - Gather facts about availability zones in AWS.
+version_added: '2.3'
+author: 'Henrique Rodrigues (github.com/Sodki)'
+options:
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value. See \
+      U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeAvailabilityZones.html) for \
+      possible filters. Filter names and values are case sensitive. You can also use underscores (_) \
+      instead of dashes (-) in the filter keys, which will take precedence in case of conflict.
+    required: false
+    default: {}
+extends_documentation_fragment:
+    - aws
+    - ec2
+requirements: [botocore, boto3]
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Gather facts about all availabily zones
+- aws_az_facts:
+
+# Gather facts about a single availability zone
+- aws_az_facts:
+    filters:
+      zone-name: eu-west-1a
+'''
+
+RETURN = '''
+availability_zones:
+    description: >
+        Availability zones that match the provided filters. Each element consists of a dict with all the information
+        related to that available zone.
+    type: list
+    sample: "[
+        {
+            'messages': [],
+            'region_name': 'us-west-1',
+            'state': 'available',
+            'zone_name': 'us-west-1b'
+        },
+        {
+            'messages': [],
+            'region_name': 'us-west-1',
+            'state': 'available',
+            'zone_name': 'us-west-1c'
+        }
+    ]"
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec, boto3_conn
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, camel_dict_to_snake_dict, HAS_BOTO3
+
+try:
+    from botocore.exceptions import ClientError
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            filters=dict(default={}, type='dict')
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if region:
+        connection = boto3_conn(
+            module,
+            conn_type='client',
+            resource='ec2',
+            region=region,
+            endpoint=ec2_url,
+            **aws_connect_params
+        )
+    else:
+        module.fail_json(msg='region must be specified')
+
+    # Replace filter key underscores with dashes, for compatibility
+    sanitized_filters = {k.replace('_', '-'):v for k,v in module.params.get('filters').items()}
```

Comment:
CI failure due to use of dict comprehensions, which aren't supported on python 2.6:

```
2017-02-17 15:34:06 Compiling ./lib/ansible/modules/cloud/amazon/aws_az_facts.py ...
2017-02-17 15:34:06 SyntaxError: ('invalid syntax', ('./lib/ansible/modules/cloud/amazon/aws_az_facts.py', 116, 50, "    sanitized_filters = {k.replace('_', '-'):v for k,v in module.params.get('filters').items()}\n"))
```

See here for an alternative: http://stackoverflow.com/questions/21069668/alternative-to-dict-comprehension-prior-to-python-2-7